File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/relicRecovery2017/basicTeleOp.java
package org.firstinspires.ftc.teamcode.relicRecovery2017;

import com.qualcomm.robotcore.eventloop.opmode.Disabled;
import com.qualcomm.robotcore.eventloop.opmode.OpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
@Disabled
@TeleOp(name ="teleop")
public class basicTeleOp extends OpMode {

    /* Declare OpMode members. */
    BaseMecanumHardware robot = new BaseMecanumHardware();

    @Override
    public void init() {
        robot.init(hardwareMap, telemetry);

        telemetry.addData("Status", "waiting to start");
        telemetry.update();
    }

    @Override
    public void loop() {
        // Run the wheels in POV mode (note: The joystick goes negative when pushed forwards so negate it)
        // In this mode the Left stick moves the robot and the Right stick rotates it left and right
        double xAxis = gamepad1.left_stick_x;
        double yAxis = -gamepad1.left_stick_y;
        double rotation = gamepad1.right_stick_x /* * ((Math.abs(yAxis) > Math.abs(xAxis))? Math.abs(yAxis): Math.abs(xAxis))*/;

        robot.moveRobot(xAxis, yAxis, rotation);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/kotlinTestCode/basicTeleOp.kt
package org.firstinspires.ftc.teamcode.kotlinTestCode

import com.cout970.vector.impl.Vector2d
import com.qualcomm.robotcore.eventloop.opmode.OpMode
import com.qualcomm.robotcore.eventloop.opmode.TeleOp

/**
 * Created by caleb on 5/30/2017.
 * This is a basic teleop OpMode coded in kotlin.
 */
@TeleOp(name = "kotlin mecanum teleop", group = "kotlin")
class BasicTeleOp : OpMode() {

    /* Declare OpMode members. */
    var robot = BaseMecanumHardware()

    override fun init() {
        robot.init(hardwareMap, telemetry)

        telemetry.addData("Status", "waiting to start")
        telemetry.update()
    }

    override fun loop() {
        // Run the wheels in POV mode (note: The joystick goes negative when pushed forwards, so negate it)
        // In this mode the Left stick moves the robot and the Right stick rotates it left and right
        var direction = Vector2d(gamepad1.left_stick_x.toDouble(), -gamepad1.left_stick_y.toDouble())
        val rotation = gamepad1.right_stick_x.toDouble()

        robot.moveRobot(direction, rotation)
    }
}